### DISCUSSION TITLE
[Brainstorming] ruby-bunder-2.2.16 and ruby-2.7.2 conflicts with possible workaround.

### DISCUSSION THREAD

**Discussion by net-wayfarer** [BlackArch/blackarch#3078] [Ideas]
**Problem:**
The recent version of [ruby-bundler as provided by Arch Linux](https://github.com/archlinux/svntogit-community/commit/89922442da7fc9d68095c706b2db338de53dee6d#diff-37538beb61ff63edebbf735dfcf39e5d732f49183d6beb097169d971875ca422) contains a conflicting file that is already provided by [ruby, also from Arch Linux repository](https://github.com/archlinux/svntogit-packages/commit/dd4714a6d98bf9843746b052aeb2f73b40c151b7#diff-3e341d2d9c67be01819b25b25d5e53ea3cdf3a38d28846cda85a195eb9b7203a).
```
error: failed to commit transaction (conflicting files)
ruby-bundler: /usr/share/man/man5/gemfile.5.gz exists in filesystem (owned by ruby)
```

Unfortunately `ruby-bundler` is required by a number of pentesting packages:
```
$ pacman -Qi ruby-bundler | grep -i required
Required By     : automato  beef  binproxy  brakeman  catphish  cewl  ciphr  cmsscanner  dawnscanner  dradis-ce  fingerprinter  glue  hoper  joomlavs  kautilya  metasploit  ruby-multi_xml  sitediff  smbexec  ssrf-proxy  tlspretense  twofi  vane  vcsmap  whatweb  whitewidow  wordpress-exploit-framework  wpbrute-rpc  wpscan  xspear  yasuo
```

**Planned workaround:**
Considering ruby is most likely to be [updated very soon, it is marked out of date, and with a new one in testing repository as of writing](https://archlinux.org/packages/extra/x86_64/ruby/). Here is a workaround I am brainstorming.

* [PKGBUILD](https://paste.rs/pJK)
* [ruby-bundler.install](https://paste.rs/9dO)

TL;DR, is PKGBUILD has slight `pkgrel` bump and includes the line `install=ruby-bundler.install`. ruby-bundler.install includes changes to relocate the contents of ruby-bundler elsewhere during any of the operations, following hints from [here](https://bbs.archlinux.org/viewtopic.php?id=39903), and *should* preserve the contents of ruby's manpage, which is currently gemfile.5.gz.

Two things to note here:
* Please **do not** add these to your Black Arch installation. Whilst it has been tested and that it works as intended, I am **not** liable for any loss incurred.
* You will still need to explicitly use the `--overwrite=` feature;
```
$ pacman -U /var/cache/pacman/aur/ruby-bundler-2.2.16-3-any.pkg.tar.xz --overwrite='/usr/share/man/man5/gemfile.5.gz'
```

Feedback? Questions? I am all ears. :)

**Comment by noraj** [comment]
I would have done `pacman -Syu --overwrite /usr/share/man/man5/gemfile.5.gz`?

**Reply by net-wayfarer** [reply]
Hi thanks for the response.

Yeah that is a simpler workaround, but I personally wanted to avoid that as it can create conjoined packages. This can create a cascading effect with the need to use `--overwrite` for whenever the ruby package(s) are updated, and any other immediate dependencies on those files for instance.

**Comment by Edu4rdSHL** [comment]
Well, both packages are from ArchLinux, I'm pretty sure that they will fix it soon.

**Reply by net-wayfarer** [reply]
True. Well I have been off for days and it has been addressed. Gonna close this.

**Comment by aancw** [comment]
Keep in mind that not everyone using testing repository. The fix for this conflicting file is just --overwrite=usr/*
